packages/amplify-category-api/src/provider-utils/awscloudformation/utils/authConfigToAppSyncAuthTypeBiDiMapper.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  AppSyncAuthType,
+  AppSyncAPIKeyAuthType,
+  AppSyncCognitoUserPoolsAuthType,
+  AppSyncOpenIDConnectAuthType,
+} from 'amplify-headless-interface';
+import _ from 'lodash';
+
+export const authConfigToAppSyncAuthType = (authConfig: any = {}): AppSyncAuthType => {
+  return _.get(authConfigToAppSyncAuthTypeMap, authConfig.authenticationType, () => undefined)(authConfig);
+};
+
+export const appSyncAuthTypeToAuthConfig = (authType?: AppSyncAuthType) => {
+  if (!authType) return undefined;
+  return _.get(appSyncAuthTypeToAuthConfigMap, authType.mode, () => undefined)(authType);
+};
+
+const authConfigToAppSyncAuthTypeMap: Record<string, (oldAuthObj: any) => AppSyncAuthType> = {
+  API_KEY: oldAuthObj => ({
```

Comment:
What do you refer to as oldAuthObj? Legacy?

Reply:
it's the authConfig object that comes out of the existing walkthrough. I'll change this variable to just "authConfig"